pyiceberg/catalog/bigquery_metastore.py

```diff
@@ -0,0 +1,409 @@
+                iceberg_tables.append((database_name, bq_table_list_item.table_id))
+        except NotFound:
+            raise NoSuchNamespaceError(f"Namespace (dataset) '{database_name}' not found.") from None
+        return iceberg_tables
+
+    def list_namespaces(self, namespace: Union[str, Identifier] = ()) -> List[Identifier]:
+        # Since this catalog only supports one-level namespaces, it always returns an empty list unless
+        # passed an empty namespace to list all namespaces within the catalog.
+        if namespace:
+            raise NoSuchNamespaceError(f"Namespace (dataset) '{namespace}' not found.") from None
+
+        # List top-level datasets
+        datasets_iterator = self.client.list_datasets()
+        return [(dataset.dataset_id,) for dataset in datasets_iterator]
+
+    def register_table(self, identifier: Union[str, Identifier], metadata_location: str) -> Table:
+        """Register a new table using existing metadata.
+
+        Args:
+            identifier Union[str, Identifier]: Table identifier for the table
+            metadata_location str: The location to the metadata
+
+        Returns:
+            Table: The newly registered table
+
+        Raises:
+            TableAlreadyExistsError: If the table already exists
+        """
+        dataset_name, table_name = self.identifier_to_database_and_table(identifier)
+
+        dataset_ref = DatasetReference(project=self.project_id, dataset_id=dataset_name)
+
+        io = self._load_file_io(location=metadata_location)
+        file = io.new_input(metadata_location)
+        metadata = FromInputFile.table_metadata(file)
+
+        try:
+            table = self._make_new_table(metadata, metadata_location, TableReference(dataset_ref=dataset_ref, table_id=table_name))
+            self.client.create_table(table)
+        except Conflict as e:
+            raise TableAlreadyExistsError(f"Table {table_name} already exists") from e
+
+        return self.load_table(identifier=identifier)
+
+    def list_views(self, namespace: Union[str, Identifier]) -> List[Identifier]:
+        raise NotImplementedError
+
+    def drop_view(self, identifier: Union[str, Identifier]) -> None:
+        raise NotImplementedError
+
+    def view_exists(self, identifier: Union[str, Identifier]) -> bool:
+        raise NotImplementedError
+
+    def load_namespace_properties(self, namespace: Union[str, Identifier]) -> Properties:
+        dataset_name = self.identifier_to_database(namespace)
+
+        try:
+            dataset = self.client.get_dataset(DatasetReference(project=self.project_id, dataset_id=dataset_name))
+
+            if dataset and dataset.external_catalog_dataset_options:
+                return dataset.external_catalog_dataset_options.to_api_repr()
+        except NotFound as e:
+            raise NoSuchNamespaceError(f"Namespace {namespace} not found") from e
+        return {}
+
+    def update_namespace_properties(
```

Comment:
In Java Catalog interface does not have update properties but set properties also update them. Is this function python has ?